tests/FailableMemoryAllocatorTest.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2007, Michael Feathers, James Grenning and Bas Vodde
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the <organization> nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE EARLIER MENTIONED AUTHORS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL <copyright holder> BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "CppUTest/TestHarness.h"
+#include "CppUTest/FailableMemoryAllocator.h"
+#include "CppUTest/TestTestingFixture.h"
+
+
+// Allocator must be global. Otherwise, it does not exist when memory leak detector
+// reports memory leaks.
+static FailableMemoryAllocator failableMallocAllocator("Failable Malloc Allocator", "malloc", "free");
+static FailableMemoryAllocator failableNewAllocator("Failable New Allocator", "new", "delete");
+static FailableMemoryAllocator failableNewArrayAllocator("Failable New [] Allocator", "new []", "delete []");
+
+
+#if CPPUTEST_USE_MEM_LEAK_DETECTION
+
+TEST_GROUP(FailableMemoryAllocator)
+{
+    void setup()
+    {
+        failableMallocAllocator.clearFailedAllocations();
+        failableNewAllocator.clearFailedAllocations();
+        failableNewArrayAllocator.clearFailedAllocations();
```

Comment:
I guess you are right. Although I ran into lots of issues (e.g. clang new overloading bug with -O2) while testing these, they were all such that are already tested elsewhere in CppUTest. I think the FailableMemoryAllocator is fully covered by just malloc tests. I can remove these.